app/assets/javascripts/dashboard.js

```diff
@@ -33,14 +33,40 @@ $(function () {
     map.touchZoomRotate.disableRotation();
     map.keyboard.disableRotation();
 
-    $("[data-user]").each(function () {
-      const user = $(this).data("user");
-      if (user.lon && user.lat) {
-        OSM.MapLibre.getMarker({ icon: "dot", color: user.color })
-          .setLngLat([user.lon, user.lat])
+    const markerObjects = $("[data-user]")
+      .map(function () {
+        const user = $(this).data("user");
+        const { lat, lon } = user;
+
+        if (!lat || !lon) return null;
+
+        const marker = OSM.MapLibre.getMarker({
+          icon: "dot",
+          color: user.color
+        })
+          .setLngLat([lon, lat])
           .setPopup(OSM.MapLibre.getPopup(user.description))
           .addTo(map);
-      }
-    });
+
+        return { marker, lat, lon };
+      })
+      .get();
+
+    const updateZIndex = () => {
+      markerObjects.forEach((item) => {
+        item.currentY = map.project([item.lon, item.lat]).y;
+      });
+      markerObjects.sort((a, b) => a.currentY - b.currentY);
+      markerObjects.forEach((item, index) => {
```

Comment:
Nit: It looks like other files are using for-of loops instead of `forEach`. Maybe we should be consistent for readability.